### PR TITLE
Liveness probe for namenodes

### PR DESCRIPTION
**Pull request by mdrio**
This PR adds liveness probe for namenodes. It aims to solve the problem of having both namenodes in standby.

### CODE REVIEW

File: charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml [hunk 1]
```diff
@@ -194,6 +200,14 @@ spec:
             name: fs
           - containerPort: {{ .Values.global.nameNodeHttpPort }}
             name: http
+
+          livenessProbe:
+            exec:
+              command:
+              - /nn-scripts/check-active.sh
+            initialDelaySeconds: 60
+            periodSeconds: 600
```
Comment (simleo) [inline charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml]: The command takes about two seconds, so perhaps `periodSeconds` can be brought down to `300` without impacting performance too much. Should we also perform this check as a readiness probe? For that one `periodSeconds` can probably be substantially lower (e.g., `60`)

File: charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml [hunk 2]
```diff
@@ -194,6 +200,14 @@ spec:
             name: fs
           - containerPort: {{ .Values.global.nameNodeHttpPort }}
             name: http
+
+          livenessProbe:
+            exec:
+              command:
+              - /nn-scripts/check-active.sh
```
Comment (ilveroluca) [inline charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml]: Il comando `hdfs haadmin -getAllServiceState | grep active` forse si potrebbe mettere direttamente quì, evitando di creare lo script `check-active.sh`, tipo
```
exec:
  command:
    - /bin/sh
    - -c
    - "hdfs haadmin -getAllServiceState | grep active"
```

Either way, LGTM

**Review by ilveroluca — APPROVED** [review]

File: charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml (hunk 2 above)
Comment (mdrio) [reply charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml]: Ok

File: charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml (hunk 1 above)
Comment (mdrio) [reply charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml]: I set periodSeconds to 200 and failureThreshold to 2

**Pull request merged**